### ISSUE TITLE
Add video transcription from audio track

### ISSUE THREAD

**Issue by curdriceaurora** [QiuYannnn/Local-File-Organizer#80]
Parent: #2

# Task 005: Add video transcription from audio track

## Description
Extract audio tracks from video files and use Distil-Whisper for accurate transcription. Synchronize transcription with video timeline to enable content-based video organization and searchability. This builds on the multi-frame analysis (Task 004) to provide comprehensive video understanding.

## Objectives
- Extract audio track from various video formats (MP4, AVI, MOV, MKV)
- Integrate Distil-Whisper model for accurate transcription
- Sync transcription timestamps with video timeline
- Handle videos without audio gracefully

### Audio Extraction
- Use ffmpeg-python to extract audio stream
- Convert to format suitable for Whisper (WAV, 16kHz)
- Handle multiple audio tracks (select primary/default)

### Transcription Pipeline
- Use Distil-Whisper large-v3 for accuracy
- Implement chunking for long videos (>30 minutes)
- Generate timestamped transcript segments
- Detect language automatically

### Timeline Synchronization
- Align transcript timestamps with scene boundaries (from Task 004)
- Create searchable index of video content
- Store transcript with video metadata

### Required Libraries
```python
# audio extraction
import ffmpeg
from pathlib import Path

# transcription
from transformers import pipeline
import torch

# timestamp handling
from datetime import timedelta
```

### Core Components
1. **AudioExtractor** - Extract audio from video files
2. **VideoTranscriber** - Transcribe audio using Distil-Whisper
3. **TimestampAligner** - Sync transcription with video timeline
4. **TranscriptStorage** - Save and index transcripts

### File Locations
- `file_organizer_v2/src/file_organizer/services/video_transcriber.py` - Video transcription service
- `file_organizer_v2/src/file_organizer/utils/audio_extractor.py` - Audio extraction utilities
- `file_organizer_v2/src/file_organizer/models/whisper_model.py` - Whisper model wrapper

## Acceptance Criteria
- [ ] Audio extraction works for all major video formats (MP4, AVI, MOV, MKV, WebM)
- [ ] Transcription accuracy >90% for clear audio
- [ ] Timestamps accurately synced with video timeline (±1 second)
- [ ] Handles videos without audio without crashing
- [ ] Processes 1 hour video in <10 minutes on CPU
- [ ] Transcripts stored in searchable format (JSON/SQLite)
- [ ] Unit tests cover audio extraction and transcription pipeline
- [ ] Integration tests verify end-to-end workflow

## Testing Strategy
- Unit tests for audio extraction with mock videos
- Transcription accuracy tests with sample videos
- Performance benchmarks for various video lengths
- Error handling tests (no audio, corrupted files)
- Integration with scene detection (Task 004)

## Dependencies
- **Task 001**: Audio file support (uses same Distil-Whisper model)
- **Task 004**: Multi-frame video analysis (provides scene timestamps)

## Related Tasks
- Task 001: Implement audio transcription for audio files
- Task 004: Implement multi-frame video analysis (scene detection)
- Task 006: Enhance video metadata extraction (uses transcript for categorization)

## Estimated Effort
- Size: **M** (Medium)
- Hours: **16**
- Parallel execution: **No** (depends on Tasks 001 and 004)

## Notes
- Distil-Whisper large-v3 is recommended for best accuracy
- Consider GPU acceleration for faster transcription
- Transcript storage format should be standardized with Task 001
- May need to handle multiple languages in future
- Audio quality significantly impacts transcription accuracy
- Consider storing raw audio separately for re-processing if needed
